content/flux/v0.x/data-types/dynamic.md

```diff
@@ -0,0 +1,622 @@
+`uint()` returns the [UInteger equivalent](/flux/v0.x/data-types/basic/uint/#convert-data-types-to-uintegers)
+of the dynamic value.
+
+```js
+import "experimental/dynamic"
+
+dynamicValue = dynamic.dynamic(v: "12")
+
+uint(v: dynamicValue)
+// Returns 12
+```
+
+{{% /tab-content %}}
+{{% tab-content %}}
+
+Use [`float()`](/flux/v0.x/stdlib/universe/float/) to convert a dynamic type to a floating point value.
+`float()` returns the [float equivalent](/flux/v0.x/data-types/basic/float/#convert-data-types-to-floats)
+of the dynamic value.
+
+```js
+import "experimental/dynamic"
+
+dynamicValue = dynamic.dynamic(v: "12.1")
+
+float(v: dynamicValue)
+// Returns 12.1
+```
+
+{{% /tab-content %}}
+{{% tab-content %}}
+
+Use [`bool()`](/flux/v0.x/stdlib/universe/bool/) to convert a dynamic type to a boolean value.
+`bool()` returns the [boolean equivalent](/flux/v0.x/data-types/basic/bool/#convert-data-types-to-booleans)
+of the dynamic value.
+
+```js
+import "experimental/dynamic"
+
+dynamicValue = dynamic.dynamic(v: "true")
+
+bool(v: dynamicValue)
+// Returns true
+```
+
+{{% /tab-content %}}
+{{% tab-content %}}
+
+Use [`duration()`](/flux/v0.x/stdlib/universe/duration/) to convert a dynamic type to a duration value.
+`duration()` returns the [duration equivalent](/flux/v0.x/data-types/basic/duration/#convert-data-types-to-durations)
+of the dynamic value.
+
+```js
+import "experimental/dynamic"
+
+dynamicValue = dynamic.dynamic(v: 3000000000)
+
+duration(v: dynamicValue)
+// Returns 3s
+```
+
+{{% /tab-content %}}
+{{% tab-content %}}
+
+Use [`time()`](/flux/v0.x/stdlib/universe/time/) to convert a dynamic type to a time value.
+`time()` returns the [time equivalent](/flux/v0.x/data-types/basic/time/#convert-data-types-to-time)
+of the dynamic value.
+
+```js
+import "experimental/dynamic"
+
+dynamicValue = dynamic.dynamic(v: 1640995200000000000)
+
+time(v: dynamicValue)
+// Returns 2022-01-01T00:00:00.000000000Z
+```
+
+{{% /tab-content %}}
+{{% tab-content %}}
+
+Use [`bytes()`](/flux/v0.x/stdlib/universe/bytes/) to convert a dynamic type to a byte-encoded string.
+`bytes()` returns the [bytes equivalent](/flux/v0.x/data-types/basic/bytes/#convert-data-types-to-bytes)
+of the dynamic value.
+
+```js
+import "experimental/dynamic"
+
+dynamicValue = dynamic.dynamic(v: "Hello world!")
+
+bytes(v: dynamicValue)
+// Returns 0x48656c6c6f20776f726c6421
+```
+{{% /tab-content %}}
+{{% tab-content %}}
+
+1.  Use [`string()`](/flux/v0.x/stdlib/universe/string/) to convert a dynamic type to a string.
+2.  Import the [`regexp` package](/flux/v0.x/stdlib/regexp/) and use
+    [`regexp.compile()`](/flux/v0.x/stdlib/regexp/compile/) to convert the string
+    value to a regular expression type.
+
+```js
+import "experimental/dynamic"
+import "regexp"
+
+dynamicValue = dynamic.dynamic(v: "^[abc][123]{1,}")
+stringValue = string(v: dynamicValue)
+
+regexp.compile(v: stringValue)
```

Comment:
This is one of the wrinkles of the status quo. I don't know if there's anything to add in the docs since it'll likely be distracting to talk about what _won't work_.

As things are, you can build a `dynamic.dynamic(v: /regex literal/)` but you can never just "unwrap" the already compiled regex from the dynamic.

We're in a similar situation for a few other types iirc. Function objects are one, but I think there are probably others I'm forgetting.